src/main/java/app/studnicki/ox/Menu.java

```diff
@@ -0,0 +1,69 @@
+package app.studnicki.ox;
+
+import app.studnicki.ox.config.Config;
+import app.studnicki.ox.game.Game;
+import app.studnicki.ox.game.GameBuilder;
+import app.studnicki.ox.initstate.InitState;
+import app.studnicki.ox.initstate.InitStateFactory;
+import app.studnicki.ox.ui.UserInterface;
+
+class Menu {
+
+  private final UserInterface ui;
+  private boolean exit;
+
+  Menu(UserInterface ui) {
+    this.ui = ui;
+  }
+
+  void run() {
+    while (!exit) {
+      ui.welcome();
+
+      int choice = ui.menu();
+
+      switch (choice) {
+        case 1:
+          runGame();
+          break;
+        case 2:
+          runLanguageMenu();
+          break;
+        case 0:
+          exit = true;
+          break;
+      }
+    }
+  }
+
+  private void runLanguageMenu() {
+    switch (ui.changeLanguage()) {
```

Comment:
How does a switch help with OO? Why not have an enum with available languages and switch on enum? Why not reduce the complexity by making enumMap with messages and just load it (or reload it) from a resource bundle depending on language code? Why allow choosing an int if you could just ask a user to type two letters from en/pl/ja and use them to search for a language key in enumMap languages and catch the error if there is no such key reducing the complexity?